tracing-node-js-micro-service-oci.md

```diff
@@ -0,0 +1,155 @@
+---
+title: Tracing A Node.js with OCI Application Performance Monitoring and Zipkin
+parent: [tutorials]
+toc: false
+tags: [open-source, oci, nodejs]
+languages: [nodejs]
+categories: [frameworks, cloudapps]
+thumbnail: assets/ociemailimage-10.jpg
+description: Using Zipkin as an open-source tracking tool with OCI Application Performance Monitoring.  
+date: 2022-02-25 19:42
+mrm: WWMK220224P00058
+author:
+    name: Javier Mugueta
+    home: https://javiermugueta.blog/author/javiermugueta/ 
+---
+OCI Application Performance Monitoring (APM) provides a comprehensive set of features to monitor applications and diagnose performance issues.
+
+Application Performance Monitoring integrates with open-source tracing system tools (open-source tracers) such as Jaeger and **[Zipkin](https://zipkin.io)** and enables you to upload trace data. It also supports context propagation between Application Performance Monitoring agents and open-source tracers.
+
+## STEP 1: Configure APM
+
+Go to APM->Administration, click in [Create APM Domain] button and provide the information requested in popup window.
+
+
+{% imgx assets/ociapmzipkinimage-92.jpg %}
+
+{% imgx assets/ociapmzipkinimage-93.jpg %}
+
+{% imgx assets/ociapmzipkinimage-94.jpg %}
+
+## STEP 2: Grab domain details
+
+In the APM domain details you created, get the **[Data Upload Endpoint]** URL and the **auto_generated_public_datakey** values, we’ll need them in step 3.
+
+
+{% imgx assets/ociapmzipkinimage-95.jpg %}
+
+## STEP 3: Configure your Node.js app
+
+Follow the steps [here](https://github.com/openzipkin/zipkin-js) to configure Zipkin in your app. [Here's the doc](https://docs.oracle.com/en-us/iaas/application-performance-monitoring/doc/configure-open-source-tracing-systems.html) from OCI APM with detailled instructions for the OCI part. 
+
+If you want to use an example, use [this](https://github.com/openzipkin/zipkin-js-example), as it's the code we're going to use for this post. Clone the repo and edit file web/recorder.js according to the instructions below (note the changes we'll make):
+
+{% imgx assets/ociapmzipkinimage-96.jpg %}
+
+```console
+/* eslint-env browser */
+const {
+  BatchRecorder,
+  jsonEncoder: {JSON_V2}
+} = require('zipkin');
+const {HttpLogger} = require('zipkin-transport-http');
+```
+
+Replace the last line with:
+```console
```

Comment:
Add newlines before code blocks (all the way through this article)

Reply:
Should be all set now